e2e/Tests/iothub/device/FileUploadE2ETests.cs

```diff
@@ -151,7 +160,7 @@ await using (deviceClient)
 
                     try
                     {
-                        await deviceClient.CompleteFileUploadAsync(notification).ConfigureAwait(false);
+                        await deviceClient.CompleteFileUploadAsync(notification, ct).ConfigureAwait(false);
                     }
                     catch (IotHubClientException ex) when (ex.ErrorCode is IotHubClientErrorCode.ServerError)
```

Comment:
ServerError is a transient error, so this wouldn't reach the client app if it has our default retry policy applied. Instead, what will reach the client app is an OperationCanceledException in case they pass in a cancellation token.
The only way to figure out why it timed out is to look at the logs.

We wouldn't want to catch an internal OperationCanceledException and add the ServerError as the inner exception, as that would change the exception stack trace.

Any suggestion on how to address this?

Reply:
What if we turn off retry policy for this test?

Reply:
Since this is a result of a bad request and will always fail, should we map this to a 400 error instead?

Reply:
It would be a mis-match from what service returns: "{"Message":"{\"errorCode\":500001,\"trackingId\":\"49a872e111874d3ba972fa281bdc149f-G:0-TimeStamp:04/20/2023 01:05:32\",\"message\":\"InternalServerError\",\"timestampUtc\":\"2023-04-20T01:05:32.2490174Z\"}","ExceptionMessage":""}"

Reply:
> What if we turn off retry policy for this test?

We could, but how would we recommend users handle this scenario?